src/policies.rs

```diff
@@ -355,6 +368,73 @@ fn verify_different_authors<G: Git>(
     }
 }
 
+fn verify_rebased<G: Git>(
+    commits: &[Commit],
+    git: &G,
+    old_commit_id: Oid,
+    new_commit_id: Oid,
+    ref_name: &str,
+    override_tag_pattern: &Option<String>,
+) -> Result<PolicyResult, Box<dyn Error>> {
+    let new_branch = old_commit_id.is_zero();
```

Comment:
Could we create a new function ie. git.is_new_branch to encapsulate this?

Reply:
I agree we probably want to encapsulate this logic somewhere. What bothers me about making it a function is that this logic to go from Oid to information about branches is very context specific. In other contexts, a zero Oid could mean something completely different.

Maybe a better approach is to pull some of these properties into some sort of metadata struct that can be passed around? Somewhere like at the entrypoint in lib.rs, since this particular Oid being zero meaning it's a new branch is part of that interface's definition?

Reply:
@danielc777888 I've taken a different approach to encapsulating this. The old commit, new commit thing is now an ADT by the time it gets to the policy. Please take a look at `ReferenceUpdate`.